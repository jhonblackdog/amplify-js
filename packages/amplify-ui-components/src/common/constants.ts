// Namespace
export const AMPLIFY_UI_PREFIX = 'amplify-ui';

// Classnames
export const AMPLIFY_UI_TEXT_INPUT = `${AMPLIFY_UI_PREFIX}-text-input`;
export const AMPLIFY_UI_SCENE = `${AMPLIFY_UI_PREFIX}-scene`;
export const AMPLIFY_UI_SCENE_LOADING = `${AMPLIFY_UI_PREFIX}-scene-loading`;

// Dictionaries

/*
  The below constants will be moved to an i18n solution.
  Storing here to have a common place to pull from once we pick up that work.
*/
// Username common constants
export const USERNAME_SUFFIX = 'username';
export const USERNAME_LABEL = 'Username *';
export const USERNAME_PLACEHOLDER = 'Enter your username';

// Email common constants
export const EMAIL_SUFFIX = 'email';
export const EMAIL_LABEL = 'Email Address *';
export const EMAIL_PLACEHOLDER = 'amplify@example.com';

// Code common constants
export const CODE_SUFFIX = 'code';
export const CODE_LABEL = 'Verification code';
export const CODE_PLACEHOLDER = 'Enter code';
export const SEND_CODE = 'Send Code';
export const CONFIRM = 'Confirm';

// Phone common constants
export const PHONE_SUFFIX = 'phone';
export const PHONE_LABEL = 'Phone Number *';
export const PHONE_PLACEHOLDER = '(555) 555-1212';

// Password common constants
export const FORGOT_PASSWORD_TEXT = 'Forgot your password?';
export const RESET_PASSWORD_TEXT = 'Reset password';
export const RESET_YOUR_PASSWORD = 'Reset your password';
export const PASSWORD_SUFFIX = 'password';
export const PASSWORD_LABEL = 'Password *';
export const PASSWORD_PLACEHOLDER = 'Enter your password';
export const BACK_TO_SIGN_IN = 'Back to Sign In';

// Country Dial Code common constants
export const COUNTRY_DIAL_CODE_SUFFIX = 'country-dial-code-select';
export const COUNTRY_DIAL_CODE_DEFAULT = '+1';

// Sign In common constants
// TODO: Change to SIGN_IN_HEADER_TEXT
export const HEADER_TEXT = 'Sign into your account';
// TODO: Change to SIGN_IN_SUBMIT_BUTTON_TEXT
export const SUBMIT_BUTTON_TEXT = 'Sign in';
export const CREATE_ACCOUNT_TEXT = 'Create account';
export const NO_ACCOUNT_TEXT = 'No account?';
export const CONFIRM_SMS_CODE = 'Confirm SMS Code';
export const CONFIRM_TOTP_CODE = 'Confirm TOTP Code';
export const SIGN_IN_WITH_AMAZON = 'Sign In with Amazon';
export const SIGN_IN_WITH_AUTH0 = 'Sign In with Auth0';
export const SIGN_IN_WITH_FACEBOOK = 'Sign In with Facebook';
export const SIGN_IN_WITH_GOOGLE = 'Sign In with Google';
export const SIGN_IN_WITH_AWS = 'Sign in with AWS';

// Sign Up common constants
export const SIGN_UP_HEADER_TEXT = 'Create a new account';
export const SIGN_UP_SUBMIT_BUTTON_TEXT = 'Create account';
export const HAVE_ACCOUNT_TEXT = 'Have an account?';
export const SIGN_IN_TEXT = 'Sign in';
export const SIGN_UP_USERNAME_PLACEHOLDER = 'Create a username';
export const SIGN_UP_PASSWORD_PLACEHOLDER = 'Create a password';

// Confirm Sign Up common constants
export const CONFIRM_SIGN_UP_HEADER_TEXT = 'Confirm Sign up';
export const CONFIRM_SIGN_UP_SUBMIT_BUTTON_TEXT = 'Confirm';
export const CONFIRM_SIGN_UP_CODE_LABEL = 'Confirmation Code';
export const CONFIRM_SIGN_UP_CODE_PLACEHOLDER = 'Enter your code';
export const CONFIRM_SIGN_UP_LOST_CODE = 'Lost your code?';
export const CONFIRM_SIGN_UP_RESEND_CODE = 'Resend Code';

// Sign Out
export const SIGN_OUT = 'Sign Out';

// Auth Keys
export const AUTH_SOURCE_KEY = 'amplify-auth-source';

// Error message Common Constants
export const PHONE_EMPTY_ERROR_MESSAGE = 'Phone number can not be empty';
export const NO_AUTH_MODULE_FOUND = 'No Auth module found, please ensure @aws-amplify/auth is imported';
