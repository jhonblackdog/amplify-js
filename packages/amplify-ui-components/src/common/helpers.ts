import { cx } from 'emotion';

// TODO: rename
export const styleNuker = (override: boolean, cn: string, ecn: string): string => override ? cn : cx(ecn, cn);
