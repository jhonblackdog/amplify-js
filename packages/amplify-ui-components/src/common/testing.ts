export const pixelThreshold = 10;
