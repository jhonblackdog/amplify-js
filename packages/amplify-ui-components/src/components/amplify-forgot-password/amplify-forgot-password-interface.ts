export type CodeDeliveryType = 'SMS' | 'EMAIL';
