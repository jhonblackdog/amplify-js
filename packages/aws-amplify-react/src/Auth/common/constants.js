export const Constants = {
    AUTH_SOURCE_KEY: 'amplify-react-auth-source',
    AUTH0: 'auth0',
    GOOGLE: 'google',
    FACEBOOK: 'facebook',
    AMAZON: 'amazon',
    SIGN_IN_WITH_HOSTEDUI_KEY: 'amplify-react-signin-with-hostedUI'
};
