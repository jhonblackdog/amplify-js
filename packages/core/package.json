{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
