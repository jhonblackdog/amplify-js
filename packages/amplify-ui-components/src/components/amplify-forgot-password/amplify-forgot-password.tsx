import { Auth } from '@aws-amplify/auth';
import { I18n, Logger } from '@aws-amplify/core';
import { Component, Prop, State, h } from '@stencil/core';

import { FormFieldTypes, PhoneNumberInterface } from '../amplify-auth-fields/amplify-auth-fields-interface';
import { AuthState, AuthStateHandler, UsernameAliasStrings } from '../../common/types/auth-types';
import {
  NO_AUTH_MODULE_FOUND,
  COUNTRY_DIAL_CODE_DEFAULT,
  PHONE_SUFFIX,
  COUNTRY_DIAL_CODE_SUFFIX,
} from '../../common/constants';
import { Translations } from '../../common/Translations';
import { CodeDeliveryType, ForgotPasswordAttributes } from './amplify-forgot-password-interface';

import {
  dispatchToastHubEvent,
  dispatchAuthStateChangeEvent,
  composePhoneNumberInput,
  checkUsernameAlias,
} from '../../common/helpers';

const logger = new Logger('ForgotPassword');

@Component({
  tag: 'amplify-forgot-password',
  shadow: true,
})
export class AmplifyForgotPassword {
  /** The header text of the forgot password section */
  @Prop() headerText: string = I18n.get(Translations.RESET_YOUR_PASSWORD);
  /** The text displayed inside of the submit button for the form */
  @Prop() submitButtonText: string = I18n.get(Translations.SEND_CODE);
  /** The form fields displayed inside of the forgot password form */
  @Prop() formFields: FormFieldTypes | string[] = [];
  /** The function called when making a request to reset password */
  @Prop() handleSend: (event: Event) => void = event => this.send(event);
  /** The function called when submitting a new password */
  @Prop() handleSubmit: (event: Event) => void = event => this.submit(event);
  /** Auth state change handler for this component */
  @Prop() handleAuthStateChange: AuthStateHandler = dispatchAuthStateChangeEvent;
  /** Username Alias is used to setup authentication with `username`, `email` or `phone_number`  */
  @Prop() usernameAlias: UsernameAliasStrings = 'username';
  @State() delivery: CodeDeliveryType | null = null;
  @State() loading: boolean = false;

  private phoneNumber: PhoneNumberInterface = {
    countryDialCodeValue: COUNTRY_DIAL_CODE_DEFAULT,
    phoneNumberValue: null,
  };
  private newFormFields: FormFieldTypes | string[] = [];

  @State() forgotPasswordAttrs: ForgotPasswordAttributes = {
    userInput: '',
    password: '',
    code: '',
  };

  componentWillLoad() {
    checkUsernameAlias(this.usernameAlias);
    if (this.formFields.length === 0) {
      switch (this.usernameAlias) {
        case 'email':
          this.newFormFields = [
            {
              type: 'email',
              required: true,
              handleInputChange: this.handleFormFieldInput('email'),
              inputProps: {
                'data-test': 'forgot-password-email-input',
              },
            },
          ];
          break;
        case 'phone_number':
          this.newFormFields = [
            {
              type: 'phone_number',
              required: true,
              handleInputChange: this.handleFormFieldInput('phone_number'),
              inputProps: {
                'data-test': 'forgot-password-phone-number-input',
              },
            },
          ];
          break;
        case 'username':
        default:
          this.newFormFields = [
            {
              type: 'username',
              required: true,
              handleInputChange: this.handleFormFieldInput('username'),
              inputProps: {
                'data-test': 'forgot-password-username-input',
              },
            },
          ];
          break;
      }
    } else {
      this.formFields.forEach(field => {
        const newField = { ...field };
        newField['handleInputChange'] = event => this.handleFormFieldInputWithCallback(event, field);
        this.newFormFields.push(newField);
      });
    }
  }

  handleFormFieldInput(fieldType) {
    switch (fieldType) {
      case 'username':
      case 'email':
        return event => (this.forgotPasswordAttrs.userInput = event.target.value);
      case 'phone_number':
        return event => this.handlePhoneNumberChange(event);
      case 'password':
      case 'code':
        return event => (this.forgotPasswordAttrs[fieldType] = event.target.value);
      default:
        return;
    }
  }

  handleFormFieldInputWithCallback(event, field) {
    const fnToCall = field['handleInputChange']
      ? field['handleInputChange']
      : (event, cb) => {
          cb(event);
        };
    const callback =
      field.type === 'phone_number'
        ? event => (this.forgotPasswordAttrs.userInput = event.target.value)
        : this.handleFormFieldInput(field.type);
    fnToCall(event, callback.bind(this));
  }

  handlePhoneNumberChange(event) {
    const name = event.target.name;
    const value = event.target.value;

    /** Cognito expects to have a string be passed when signing up. Since the Select input is separate
     * input from the phone number input, we need to first capture both components values and combined
     * them together.
     */

    if (name === COUNTRY_DIAL_CODE_SUFFIX) {
      this.phoneNumber.countryDialCodeValue = value;
    }

    if (name === PHONE_SUFFIX) {
      this.phoneNumber.phoneNumberValue = value;
    }
  }

  async send(event) {
    if (event) {
      event.preventDefault();
    }
    if (!Auth || typeof Auth.forgotPassword !== 'function') {
      throw new Error(NO_AUTH_MODULE_FOUND);
    }
    this.loading = true;

    switch (this.usernameAlias) {
      case 'phone_number':
        this.forgotPasswordAttrs.userInput = composePhoneNumberInput(this.phoneNumber);
        break;
      default:
        break;
    }

    try {
      const data = await Auth.forgotPassword(this.forgotPasswordAttrs.userInput);
      logger.debug(data);
      this.newFormFields = [
        {
          type: 'code',
          required: true,
          handleInputChange: this.handleFormFieldInput('code'),
          inputProps: {
            'data-test': 'forgot-password-code-input',
          },
        },
        {
          type: 'password',
          required: true,
          handleInputChange: this.handleFormFieldInput('password'),
          label: 'New password',
          placeholder: 'Enter your new password',
        },
      ];
      this.delivery = data.CodeDeliveryDetails;
    } catch (error) {
      dispatchToastHubEvent(error);
    } finally {
      this.loading = false;
    }
  }

  async submit(event: Event) {
    if (event) {
      event.preventDefault();
    }
    if (!Auth || typeof Auth.forgotPasswordSubmit !== 'function') {
      throw new Error(NO_AUTH_MODULE_FOUND);
    }
    this.loading = true;
    try {
      const { userInput, code, password } = this.forgotPasswordAttrs;
      const data = await Auth.forgotPasswordSubmit(userInput, code, password);
      logger.debug(data);
      this.handleAuthStateChange(AuthState.SignIn);
      this.delivery = null;
    } catch (error) {
      dispatchToastHubEvent(error);
    } finally {
      this.loading = false;
    }
  }

  render() {
    const submitFn = this.delivery ? event => this.handleSubmit(event) : event => this.handleSend(event);
    return (
      <amplify-form-section
        headerText={this.headerText}
        handleSubmit={submitFn}
        loading={this.loading}
        secondaryFooterContent={
          <amplify-button
            variant="anchor"
            onClick={() => this.handleAuthStateChange(AuthState.SignIn)}
            data-test="forgot-password-back-to-sign-in-link"
          >
            {I18n.get(Translations.BACK_TO_SIGN_IN)}
          </amplify-button>
        }
        testDataPrefix={'forgot-password'}
      >
        <amplify-auth-fields formFields={this.newFormFields} />
      </amplify-form-section>
    );
  }
}
