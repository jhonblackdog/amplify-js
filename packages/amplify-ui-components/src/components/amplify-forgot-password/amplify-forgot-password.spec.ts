import { newSpecPage } from '@stencil/core/testing';
import { AmplifyForgotPassword } from './amplify-forgot-password';
import { RESET_YOUR_PASSWORD, SEND_CODE } from '../../common/constants';

describe('amplify-forgot-password spec:', () => {
  describe('Component logic ->', () => {
    let amplifyForgotPassword;

    beforeEach(() => {
      amplifyForgotPassword = new AmplifyForgotPassword();
    });

    it('`headerText` should be set by default', () => {
      expect(amplifyForgotPassword.headerText).toBe(RESET_YOUR_PASSWORD);
    });

    it('`submitButtonText` should be set by default', () => {
      expect(amplifyForgotPassword.submitButtonText).toBe(SEND_CODE);
    });

    it('`overrideStyle` should be false by default', () => {
      expect(amplifyForgotPassword.overrideStyle).toBe(false);
    });
  });
  describe('Render logic ->', () => {
    it(`should render`, async () => {
      const page = await newSpecPage({
        components: [AmplifyForgotPassword],
        html: `<amplify-forgot-password />`,
      });
      expect(page.root).toMatchSnapshot();
    });
  });
});
