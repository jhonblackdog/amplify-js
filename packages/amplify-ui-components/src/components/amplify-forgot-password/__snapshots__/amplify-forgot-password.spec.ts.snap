// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-forgot-password spec: Render logic -> should render 1`] = `
<amplify-forgot-password>
  <amplify-form-section headertext="Reset your password">
    <amplify-auth-fields></amplify-auth-fields>
  </amplify-form-section>
</amplify-forgot-password>
`;
